llama-index-integrations/llms/llama-index-llms-mistralai/llama_index/llms/mistralai/base.py

```diff
@@ -252,6 +266,80 @@ def stream_complete(
         stream_complete_fn = stream_chat_to_completion_decorator(self.stream_chat)
         return stream_complete_fn(prompt, **kwargs)
 
+    def _get_tool_call(
+        self,
+        response: ChatResponse,
+    ) -> ToolCall:
+        tool_calls = response.message.additional_kwargs.get("tool_calls", [])
+
+        if len(tool_calls) < 1:
+            raise ValueError(
+                f"Expected at least one tool call, but got {len(tool_calls)} tool calls."
+            )
+
+        # TODO: support more than one tool call?
```

Comment:
i think mistral technically supports parallel function calling

Reply:
actually nvm

Reply:
I found (openai especially) will create two calls, and hallucinate the inputs to the second tool call lol

Reply:
hmm i see. if that's a function of the model though, wonder if we should just allow it anyways

Reply:
If we separated tool selection and tool calling into two pieces, then we could let the user decide 👀 predict_and_call could continue to assume one